pkg/build/controller/strategy/util.go

```diff
@@ -193,6 +164,25 @@ func addOriginVersionVar(output *[]kapi.EnvVar) {
 	*output = append(*output, version)
 }
 
+// addOutputEnvVars adds env variables that provide information about the output
+// target for the build
+func addOutputEnvVars(buildOutput *kapi.ObjectReference, output *[]kapi.EnvVar) error {
```

Comment:
I find it easier to reason about pure functions than functions provide output by mutating one of the arguments. Mutating a pointer argument is often needed in C, but not in Go since we can have multiple return values.

Example:

```go
// outputEnvVars return environment variables that provide information about the
// buildOutput target.
func outputEnvVars(buildOutput *kapi.ObjectReference) ([]kapi.EnvVar, error) {
	if buildOutput == nil {
		// output must always be a DockerImage type reference at this point.
		return nil, nil
	}

	ref, err := imageapi.ParseDockerImageReference(buildOutput.Name)
	if err != nil {
		return nil, err
	}
	registry := ref.Registry
	ref.Registry = ""
	image := ref.String()

	return []kapi.EnvVar{
		{Name: "OUTPUT_REGISTRY", Value: registry},
		{Name: "OUTPUT_IMAGE", Value: image},
	}, nil
}
```